### PR TITLE
[STORM-3619] Add null check for topology name

### PR DESCRIPTION
**Pull request by Ethanlm**
https://issues.apache.org/jira/browse/STORM-3619

With this change,  submitting a topology with a `null` name will fail with clear message

```
Exception in thread "main" java.lang.IllegalArgumentException: Topology name 'null' is not valid. It can't be null and it must match ^[^/.:\\]+$
```

### CODE REVIEW

File: storm-client/src/jvm/org/apache/storm/StormSubmitter.java [hunk 1]
```diff
@@ -435,7 +436,7 @@ public void onCompleted(String srcFile, String targetFile, long totalBytes) {
         });
     }
 
-    private static boolean topologyNameExists(String name, NimbusClient client) {
+    private static boolean isTopologyNameAllowed(String name, NimbusClient client) {
```
Comment (Ethanlm) [inline storm-client/src/jvm/org/apache/storm/StormSubmitter.java:439]: Changing the name here because it calls `client.getClient().isTopologyNameAllowed` and it doesn't only check the existence of the topology name.

File: storm-client/src/jvm/org/apache/storm/utils/Utils.java [hunk 2]
```diff
@@ -122,7 +122,9 @@
     // tests by subclassing.
     private static Utils _instance = new Utils();
     private static String memoizedLocalHostnameString = null;
-    public static final Pattern TOPOLOGY_KEY_PATTERN = Pattern.compile("^[\\w \\t\\._-]+$", Pattern.UNICODE_CHARACTER_CLASS);
+    public static final Pattern BLOB_KEY_PATTERN =
+            Pattern.compile("^[\\w \\t\\._-]+$", Pattern.UNICODE_CHARACTER_CLASS);
+    private static final Pattern TOPOLOGY_NAME_REGEX = Pattern.compile("^[^/.:\\\\]+$");
```
Comment (Ethanlm) [inline storm-client/src/jvm/org/apache/storm/utils/Utils.java:127]: Copied `TOPOLOGY_NAME_REGEX` from storm-server/src/main/java/org/apache/storm/daemon/nimbus/Nimbus.java. We should have a consistent check for topology name. 

`TOPOLOGY_KEY_PATTERN` was originally only used for blob key. See https://github.com/apache/storm/pull/2703

File: storm-client/src/jvm/org/apache/storm/StormSubmitter.java [hunk 3]
```diff
@@ -248,11 +252,8 @@ public static void submitTopologyAs(String name, Map<String, Object> topoConf, S
         try {
             String serConf = JSONValue.toJSONString(topoConf);
             try (NimbusClient client = NimbusClient.getConfiguredClientAs(conf, asUser)) {
-                if (topologyNameExists(name, client)) {
-                    throw new RuntimeException("Topology with name `" + name + "` already exists on cluster");
-                }
-                if (!Utils.isValidKey(name)) {
```
Comment (agresch) [inline storm-client/src/jvm/org/apache/storm/StormSubmitter.java:254]: why was this removed?

Comment (Ethanlm) [reply storm-client/src/jvm/org/apache/storm/StormSubmitter.java:254]: There are two reasons:
1. Moved the name check to the very beginning of this submit function. 
2. Replace it with `Utils.validateTopologyName(name);` so we have the same check on client and server side.
